tardis/interfaces/siteadapter.py

```diff
@@ -1,13 +1,37 @@
+from ..configuration.configuration import Configuration
 from ..utilities.attributedict import AttributeDict
 
 from abc import ABCMeta, abstractmethod
+from cobald.utility.primitives import infinity as inf
 from enum import Enum
+from functools import lru_cache
+from pydantic import BaseModel, conint, validator
+from typing import Optional
 
 import logging
 
 logger = logging.getLogger("cobald.runtime.tardis.interfaces.site")
 
 
+class SiteConfigurationModel(BaseModel):
+    """
+    pydantic BaseModel for the input validation of the generic site configuration
+    """
+
+    name: str
+    adapter: str
+    quota: Optional[int] = inf
+    drone_minimum_lifetime: Optional[conint(gt=0)] = None
```

Comment:
1. 0 would imply to directly drain a freshly starting drone. I think that is not reasonable. However, since the there is currently a 60s heartbeat interval (see #166) , the drone will be drained after `n` seconds, where `n` is the heartbeat.
2. Simple reason, I was not aware of `PositiveInt`. 

Are there other opionions on allowing 0? 
Do you want me to change the code to use `PostiveInt` instead?

Reply:
You are right, 0 indeed is not very sensible. Sorry I mixed it up with the heartbeat interval from #166.
As for `PositiveInt` or not, I won't insist on it. For me both are fine.